client/src/app/index.js

```diff
@@ -1,13 +1,43 @@
 // npm packages
-import React from 'react';
+import React, {Component} from 'react';
+import {connect} from 'react-redux';
+import {push} from 'react-router-redux';
 
 import Header from '../components/header';
 import Footer from '../components/footer';
 
-export default ({children}) => (
-  <div className="container">
-    <Header />
-    {children}
-    <Footer />
-  </div>
-);
+const mapStateToProps = state => ({
+  actualPath: state.routing.locationBeforeTransitions && state.routing.locationBeforeTransitions.pathname,
+  user: state.auth.user,
+  token: state.auth.token,
+});
+
+const mapDispatchToProps = dispatch => ({
+  navTo: (path) => dispatch(push(path)),
+});
+
+class App extends Component {
```

Comment:
1. App should be a dumb container without any logic.
2. This might result in an infinite redirect. I think using [redux-auth-wrapper](https://github.com/mjrussell/redux-auth-wrapper) for this would be a better solution.